docs/src/tutorials/random_scattering.md

```diff
@@ -0,0 +1,113 @@
+# Random scattering (deflection) of a single particle
+
+```@setup random_scattering
+import Pkg
+Pkg.activate("../../../tutorials")
+Pkg.develop(path="../../..")
+Pkg.instantiate()
+```
+
+## Intro and Setup
+In this tutorial we implement and differentiate through a single particle scattering process.
+
+In short, when a particle travels through some material with finite depth, at each step, the
+particle has a chance (!) to interact, when it does, it loses energy and randomly change direction
+[^1].
+
+```@example random_scattering
+using StochasticAD, Statistics, Distributions
+```
+
+First of all, the interaction probability is controlled by a parameter `par`, think of this as
+material property (~density), which we might be intereted in differentiate against to optimize
+material.
+
+```@example random_scattering
+
+"""
+Probability of interaction is a function of parameter we're intereted in diff. against
+"""
+function interact_prob(x,par)
+    return 1/(1+exp(-(x-par)/0.2))*0.5
+end
+```
+
+Then, we implement the "one step" in propagating particle, note instead of `if scattered
+...`, we need to do it branchlessly, we also decide when interaction happens, we lose 10% of energy.
+```@example random_scattering
+"""
+A single step of propagation, next1 and next2 are branchless "next state of particle"
+"""
+function propagate_scatter(E,phi,x,y,par)
+    x = x + cos(phi)
```

Comment:
Should we have more meaningful dynamics with a step size parameter based on a trajectory model?

Reply:
yeah maybe, anything immediately come to your mind?

Reply:
Perhaps spiralling out of the center https://math.stackexchange.com/q/3145491/85217

Reply:
ah. I was thinking if we want this to be more physical, and given our main loop is fixed time step, we can just use velocity (calculated from the remaining energy)

Reply:
That's of course also good, I just thought that you wanted something with rotation because of `sin` and `cos`